### PR TITLE
feat: add some spinners for pushing and pulling

### PR DESCRIPTION
**Pull request by yetone**

### CODE REVIEW

File: bentoml/_internal/yatai_client/__init__.py [hunk 1]
```diff
@@ -104,28 +108,57 @@ def read(*args, **kwargs):
...
+        super().__init__()
+
+
+CallbackIOWrapper: t.Type[BinaryIOCast] = t.cast(
+    t.Type[BinaryIOCast], _CallbackIOWrapper
+)
+
+
+# Just make type checker happy
+class ProgressCast(Progress):
```
Comment (bojiang) [inline bentoml/_internal/yatai_client/__init__.py:123]: use protocol here?

Comment (yetone) [reply bentoml/_internal/yatai_client/__init__.py:123]: @bojiang Do not need `t.Protocol`, this dummy class is just to fix the type annotation error of the rich library, not to add type to it.

Comment (bojiang) [reply bentoml/_internal/yatai_client/__init__.py:123]: okay

File: bentoml/_internal/yatai_client/__init__.py [hunk 2]
```diff
@@ -203,35 +241,37 @@ def push_model(model: "Model"):
...
+                            bentoml_version=info.bentoml_version,
+                            apis=apis,
+                            models=info.models,
+                        ),
+                        labels=labels,
                     ),
-                    labels=labels,
-                ),
+                )
+        with self.spin(text=f"Presign bento {bento.tag} upload url"):
```
Comment (parano) [inline bentoml/_internal/yatai_client/__init__.py:267]:
```suggestion
        with self.spin(text=f"Getting presigned upload url for {bento.tag} .."):
```

File: bentoml/_internal/yatai_client/__init__.py [hunk 3]
```diff
@@ -203,35 +241,37 @@ def push_model(model: "Model"):
...
-                req=CreateBentoSchema(
-                    description="",
-                    version=version,
-                    build_at=info.creation_time,
-                    manifest=BentoManifestSchema(
-                        service=info.service,
-                        bentoml_version=info.bentoml_version,
-                        apis=apis,
-                        models=info.models,
+            with self.spin(text=f"Creating bento {bento.tag}"):
```
Comment (parano) [inline bentoml/_internal/yatai_client/__init__.py:251]:
```suggestion
            with self.spin(text=f"Registering Bento {bento.tag} metadata.."):
```

File: bentoml/_internal/yatai_client/__init__.py [hunk 4]
```diff
@@ -203,35 +241,37 @@ def push_model(model: "Model"):
...
+                bento_repository_name=bento_repository.name, version=version
             )
-        remote_bento = yatai_rest_client.presign_bento_upload_url(
-            bento_repository_name=bento_repository.name, version=version
-        )
         with io.BytesIO() as tar_io:
             bento_dir_path = bento.path
+            if bento_dir_path is None:
+                raise CLIException(f"Bento {bento.tag} path cannot be None")
             with self.spin(text=f"Taring bento {bento.tag}"):
```
Comment (parano) [inline bentoml/_internal/yatai_client/__init__.py:275]:
```suggestion
            with self.spin(text=f'Creating tar archive for bento "{bento.tag}"..'):
```

File: bentoml/_internal/yatai_client/__init__.py [hunk 5]
```diff
@@ -546,17 +607,20 @@ def _do_pull_model(
...
             pass
         yatai_rest_client = get_current_yatai_rest_api_client()
         _tag = Tag.from_taglike(tag)
-        remote_model = yatai_rest_client.presign_model_download_url(
-            _tag.name, _tag.version
-        )
+        name = _tag.name
+        version = _tag.version
+        if version is None:
+            raise CLIException(f"Model {_tag} version cannot be None")
```
Comment (parano) [inline bentoml/_internal/yatai_client/__init__.py:619]:
```suggestion
            raise BentoMLException(f'Model "{_tag}" version cannot be None')
```

**Pull request merged**